src/MongoDB.Driver.Core/Core/Clusters/CryptClientCreator.cs

```diff
@@ -65,24 +65,21 @@ private CryptClient CreateCryptClient(CryptOptions options)
 
         private CryptOptions CreateCryptOptions()
         {
-            Dictionary<KmsType, IKmsCredentials> kmsProvidersMap = null;
+            Dictionary<KmsType, KmsCredentials> kmsProvidersMap = null;
             if (_kmsProviders != null && _kmsProviders.Count > 0)
             {
-                kmsProvidersMap = new Dictionary<KmsType, IKmsCredentials>();
-                if (_kmsProviders.TryGetValue("aws", out var awsProvider))
+                kmsProvidersMap = new Dictionary<KmsType, KmsCredentials>();
+                foreach (var kmsProvider in _kmsProviders)
                 {
-                    if (awsProvider.TryGetValue("accessKeyId", out var accessKeyId) &&
-                        awsProvider.TryGetValue("secretAccessKey", out var secretAccessKey))
+                    if (!Enum.TryParse<KmsType>(kmsProvider.Key, true, out var kmsType))
                     {
-                        kmsProvidersMap.Add(KmsType.Aws, new AwsKmsCredentials((string)secretAccessKey, (string)accessKeyId));
-                    }
-                }
-                if (_kmsProviders.TryGetValue("local", out var localProvider))
-                {
-                    if (localProvider.TryGetValue("key", out var keyObject) && keyObject is byte[] key)
-                    {
-                        kmsProvidersMap.Add(KmsType.Local, new LocalKmsCredentials(key));
+                        var message = $"The provided kms provider {kmsProvider.Key} is not supported. The list of supported providers: {string.Join(", ", Enum.GetValues(typeof(KmsType)))}.";
+                        throw new NotSupportedException(message);
                     }
+                    var kmsTypeDocumentKey = kmsProvider.Key.ToLower();
+                    var kmsProviderDocument = CreateProviderDocument(kmsTypeDocumentKey, kmsProvider.Value);
+                    var kmsCredentials = new KmsCredentials(kmsType, credentialsBytes: kmsProviderDocument.ToBson());
```

Comment:
kmsType argument is not required here. The only reason why I left it as is improving developer experience. To avoid the need to convert bytes into something understandable mostly in debugging